File: bench/Reference/sieve.js
function sieve(n) {
    const is_prime = new Uint8Array(n + 1).fill(1);
    is_prime[0] = is_prime[1] = 0;
    for (let i = 2; i * i <= n; i++)
        if (is_prime[i]) for (let j = i*i; j <= n; j += i) is_prime[j] = 0;
    return is_prime.reduce((a, b) => a + b, 0);
}
console.log(sieve(500));

File: bench/Reference/sieve.java
class sieve {
    public static void main(String[] args) {
        int n = 100000;
        boolean[] is_prime = new boolean[n + 1];
        java.util.Arrays.fill(is_prime, true);
        is_prime[0] = is_prime[1] = false;
        for (int i = 2; i * i <= n; i++)
            if (is_prime[i]) for (int j = i*i; j <= n; j += i) is_prime[j] = false;
        int count = 0;
        for (boolean b : is_prime) if (b) count++;
        System.out.println(count);
    }
}
